### PR TITLE
Fixes #1238: convert internal-only `ErrorCode`s into `ErrorCode.SERVER_INTERNAL_ERROR`

### LINKED ISSUE #1238: Replace internal (non-user-triggerable) `ErrorCode` with `ErrorCode.SERVER_INTERNAL_ERROR`
There are a few internal assertion-style failures with dedicated `ErrorCode`s: for example, `SHRED_INTERNAL_NO_PATH`. These should never occur; and if they do there are No User Serviceable Parts Inside -- no point exposing specific types externally. Instead we should use newly added `ErrorCode.SERVER_INTERNAL_ERROR` with suitable suffix we can use for debugging should ones actually occur.

List of such codes includes:

* `FILTER_UNRESOLVABLE`
* `INVALID_LOGIC_OPERATOR`
* `SHRED_INTERNAL_NO_PATH`
* `SHRED_UNRECOGNIZED_NODE_TYPE`
* `UNSUPPORTED_CQL_QUERY_TYPE`
* `UNSUPPORTED_OPERATION`

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Removes redundant "internal-only" `ErrorCode`s with `SERVER_INTERNAL_ERROR`

**Which issue(s) this PR fixes**:
Fixes #1238

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/CommandResult.java [hunk 1]
```diff
@@ -194,11 +194,9 @@ public record Error(
    * @return
    */
   public RestResponse map() {
-    if (null != this.errors() && !this.errors().isEmpty()) {
+    if (null != this.errors()) {
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/CommandResult.java:197]: Just simplifying code, not related to main changes.

File: src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java [hunk 2]
```diff
@@ -66,21 +64,16 @@ public enum ErrorCode {
...
-  SHRED_DOC_KEY_NAME_VIOLATION("Document field name invalid"),
-
   SHRED_BAD_EJSON_VALUE("Bad JSON Extension value"),
 
   SHRED_BAD_VECTOR_SIZE("$vector value can't be empty"),
 
   SHRED_BAD_VECTOR_VALUE("$vector value needs to be array of numbers"),
   SHRED_BAD_VECTORIZE_VALUE("$vectorize search clause needs to be non-blank text value"),
 
+  SHRED_DOC_KEY_NAME_VIOLATION("Document field name invalid"),
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/ErrorCode.java:74]: Moved to be alphabetically ordered (within SHRED_DOC_ section at least)

File: src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ConstraintViolationExceptionMapper.java [hunk 3]
```diff
@@ -39,8 +39,7 @@ public RestResponse<CommandResult> constraintViolationException(
         violations.stream().map(ConstraintViolationExceptionMapper::getError).distinct().toList();
 
     // return result
-    CommandResult commandResult = new CommandResult(errors);
-    return RestResponse.ok(commandResult);
+    return new CommandResult(errors).map();
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ConstraintViolationExceptionMapper.java:42]: Unifying handling; does not change behavior.

File: src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/GenericExceptionMapper.java [hunk 4]
```diff
@@ -15,6 +15,6 @@ public class GenericExceptionMapper {
   @ServerExceptionMapper({Exception.class, MismatchedInputException.class})
   public RestResponse<CommandResult> genericExceptionMapper(Throwable e) {
     CommandResult commandResult = new ThrowableCommandResultSupplier(e).get();
-    return RestResponse.ok(commandResult);
+    return commandResult.map();
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/GenericExceptionMapper.java:18]: Same as above, no change in behavior but more uniform handling.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ReadOperation.java [hunk 5]
```diff
@@ -129,7 +129,7 @@ default Uni<FindResponse> findDocument(
                           getDocumentId(row.getTupleValue(0)), // key
                           row.getUuid(1), // tx_id
                           root);
-                } catch (JsonProcessingException e) {
+                } catch (JacksonException e) {
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ReadOperation.java:132]: Still catches same actual exceptions but `JacksonException` preferred over `JsonProcessingException`.

File: src/main/java/io/stargate/sgv2/jsonapi/service/projection/DocumentProjector.java [hunk 6]
```diff
@@ -84,11 +83,8 @@ public static DocumentProjector createFromDefinition(
       return defaultProjector();
     }
     if (!projectionDefinition.isObject()) {
-      throw new JsonApiException(
-          ErrorCode.UNSUPPORTED_PROJECTION_PARAM,
-          ErrorCode.UNSUPPORTED_PROJECTION_PARAM.getMessage()
-              + ": definition must be OBJECT, was "
-              + projectionDefinition.getNodeType());
+      throw ErrorCode.UNSUPPORTED_PROJECTION_PARAM.toApiException(
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/projection/DocumentProjector.java:86]: Using newer `ErrorCode.toApiException()` as more compact, templatised variant, to make later refactoring simpler.

**Review by Hazel-Datastax — APPROVED** [review]
LGTM

**Pull request merged**